### PR TITLE
People Management: Invitations Service Layer

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:
This PR implements the Service Layer required to send invitations to a blog. To test, please, run the unit tests.

Ref. #5381

Needs review: @diegoreymendez 
Thanks in advance Diego!

### CODE REVIEW

File: WordPress/Classes/Networking/PeopleRemote.swift [hunk 1]
```diff
@@ -172,6 +172,80 @@ class PeopleRemote: ServiceRemoteWordPressComREST {
...
+            "invitees"  : usernameOrEmail,
+            "role"      : role.rawValue,
+            "message"   : message
+        ]
+
+        wordPressComRestApi.POST(path, parameters: parameters, success: { (responseObject, httpResponse) in
+            guard let responseDict = responseObject as? [String: AnyObject],
+                let validRecipients = responseDict["sent"] as? [String] else
+            {
+                completion(false)
```
Comment (diegoreymendez) [inline WordPress/Classes/Networking/PeopleRemote.swift]: I'd avoid a silent failure here.  A badly written response is not the same as an invitation not being valid to the eyes of the caller of this method.

File: WordPress/Classes/Services/PeopleService.swift [hunk 2]
```diff
@@ -148,6 +148,33 @@ struct PeopleService {
...
+    /// Sends an Invitation to a specified recipient, to access a Blog.
+    ///
+    /// - Parameters:
+    ///     - usernameOrEmail: Recipient that should be validated.
+    ///     - role: Role that would be granted to the recipient.
+    ///     - message: String that should be sent to the users.
+    ///     - completion: Closure to be executed on completion. The boolean will indicate whether the OP
+    ///       was successful, or not.
+    ///
+    func sendInvitation(usernameOrEmail: String, role: Role, message: String?, completion: (Bool -> Void)) {
```
Comment (diegoreymendez) [inline WordPress/Classes/Services/PeopleService.swift]: You could use a default value for the `message` parameter, and remove `unwrappedMessage`.

```swift
func sendInvitation(usernameOrEmail: String, role: Role, message: String = "", completion: (Bool -> Void)) {
```

**Comment by diegoreymendez** [comment]
@jleandroperez - Added 2 comments... let me know when this is ready for another look.

File: WordPress/Classes/Services/PeopleService.swift (hunk 2 above)
Comment (jleandroperez) [inline WordPress/Classes/Services/PeopleService.swift]: Much better, thanks for the suggestion!

**Comment by jleandroperez** [comment]
@diegoreymendez thanks for the review!!

Ready for another look sir!

**Comment by jleandroperez** [comment]
@diegoreymendez sorry about the second (third?) poke. I had to wrap few of the remote errors, so that we can properly localize the strings.

Ready for another look, for real this time. Thanks!

File: WordPress/Classes/Networking/PeopleRemote.swift [hunk 3]
```diff
@@ -273,4 +363,26 @@ private extension PeopleRemote {
         let filtered = parsed.filter { $0 != .Unsupported }
         return filtered.sort()
     }
+
+    /// Parses a
```
Comment (diegoreymendez) [inline WordPress/Classes/Networking/PeopleRemote.swift]: Thunderstruck! ⚡

File: WordPress/Classes/Networking/PeopleRemote.swift [hunk 4]
```diff
@@ -172,6 +175,93 @@ class PeopleRemote: ServiceRemoteWordPressComREST {
...
+        }, failure: { (error, httpResponse) in
+            failure(error)
+        })
+    }
+
+
+    /// Sends an Invitation to the specified recipient.
+    ///
+    /// - Parameters:
+    ///     - siteID: The ID of the site associated.
```
Comment (diegoreymendez) [inline WordPress/Classes/Networking/PeopleRemote.swift]: Very minor, but maybe change "site associated" to  "associated site"?

**Comment by jleandroperez** [comment]
@diegoreymendez sooooorry about that. Should be good now!

**Comment by diegoreymendez** [comment]
:shipit:

**Comment by jleandroperez** [comment]
Thank you @diegoreymendez!!

**Pull request merged**